clients/src/test/java/org/apache/kafka/test/TestUtils.java

```diff
@@ -269,4 +277,34 @@ public static void waitForCondition(final TestCondition testCondition, final lon
         }
     }
 
-}
+    /**
+     * Checks if a cluster id is valid.
+     * @param clusterId
+     * @return true if valid, false otherwise.
+     */
+    public static void isValidClusterId(String clusterId) {
+        assertNotNull(clusterId);
+
+        // Base 64 encoded value is 22 characters
+        assertEquals(clusterId.length(), 22);
+
+        Pattern clusterIdPattern = Pattern.compile("[a-zA-Z0-9_\\-]+");
+        Matcher matcher = clusterIdPattern.matcher(clusterId);
+        assertTrue(matcher.matches());
+
+        // Convert into normal variant and add padding at the end.
+        String originalClusterId = String.format("%s==", clusterId.replace("_", "/").replace("-", "+"));
+        byte[] decodedUuid = DatatypeConverter.parseBase64Binary(originalClusterId);
+
+        // We expect 16 bytes, same as the input UUID.
+        assertEquals(decodedUuid.length, 16);
+
+        //Check if it can be converted back to a UUID.
+        try {
+            ByteBuffer uuidBuffer = ByteBuffer.wrap(decodedUuid);
+            UUID.fromString(new UUID(uuidBuffer.getLong(), uuidBuffer.getLong()).toString());
```

Comment:
Is there a case where `UUID.fromString` fails when given the output of `UUID.toString`? If not, maybe it's redundant.